### PR TITLE
Filter to see students edits in particular namespace

### PR DESCRIPTION
**Pull request by vaidehi44**
## What this PR does
This PR tries to resolve this issue - #4961

## Screenshots
Before:
image

After:
image

### CODE REVIEW

**Comment by vaidehi44** [comment]
@ragesoss, I have provided options to filter for article, user and talk namespace. Let me know if it's okay, or should add more.

File: app/assets/javascripts/components/students/components/Articles/SelectedStudent/RevisionsList/StudentRevisionsList.jsx [hunk 1]
```diff
@@ -67,10 +90,27 @@ export class StudentRevisionsList extends React.Component {
...
     const keys = { recent_revisions, character_sum_ms, references_count, total_uploads };
 
+    const filterLabel = <b>Namespace Filter:</b>;
+    const filterRevisions = (
+      <select
+        className="filter-revisions"
+        value={this.state.namespace}
+        onChange={this.onNamespaceChange}
+      >
+        <option value="all">All</option>
```
Comment (ragesoss) [inline app/assets/javascripts/components/students/components/Articles/SelectedStudent/RevisionsList/StudentRevisionsList.jsx:100]: The option text for each of these should use I18n.

File: app/assets/javascripts/components/students/components/Articles/SelectedStudent/RevisionsList/StudentRevisionsList.jsx [hunk 2]
```diff
@@ -14,12 +14,26 @@ export class StudentRevisionsList extends React.Component {
...
     };
 
     this.toggleDrawer = this.toggleDrawer.bind(this);
   }
 
+  onNamespaceChange = (e) => {
+    return this.setState({ namespace: e.target.value });
+  };
+
+  namespaceToId(nm) {
```
Comment (ragesoss) [inline app/assets/javascripts/components/students/components/Articles/SelectedStudent/RevisionsList/StudentRevisionsList.jsx:28]: This is is likely a generally useful mapping that we may want to extend later on, and may want to use in other places at some point.

I suggest extracting it to `article_utils.js`.

**Comment by ragesoss** [comment]
This works smoothly when used as intended. Nice work!

There are two UX issues I noticed that might be confusing in some cases.

1. If the filter selector is used when the revisions drawer is collapsed (as it is by default) then there's no visible change when a filter option is selected. It would be better if, when the drawer is closed, changing the filter selection would open the drawer.
2. The 'no revisions' message is slightly misleading when a filter is selected such that no revisions are shown: "This editor has made no revisions.". If there are revisions in other namespaces, it's not the case that the user has made no revisions (or even no revisions in the selected namespace), just that of the 10 most recent revisions, none of them are in that namespace. Maybe the message should just be "(no revisions)".

**Comment by vaidehi44** [comment]
Okay @ragesoss. About your second point; could a more detailed message be used, like 'There are no recent edits in this namespace'? Also, the original message, 'This editor has made no revisions.', should still show if there are no revisions for `All` option, right?

**Comment by ragesoss** [comment]
@vaidehi44 those kinds of variable messages would be maybe be a tiny bit better for users, but I think the simplicity of using the same message in all situations of an empty table would be a little better because it won't add code complexity and making the message vary by filter might make the code harder to work with later on.

**Comment by vaidehi44** [comment]
Okay, will use "(no revisions)" for all the cases.

**Comment by ragesoss** [comment]
Perfect!

**Pull request merged**